Add test to verify dist_autograd doesn't populate .grad field.

pritamdamania87 opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #34162 Add worker_name helper to dist_utils.
* **#33949 Add test to verify dist_autograd doesn't populate .grad field.**

Differential Revision: [D20165254](https://our.internmc.facebook.com/intern/diff/D20165254/)

pritamdamania committed:
Add test to verify dist_autograd doesn't populate .grad field.



Differential Revision: [D20165254](https://our.internmc.facebook.com/intern/diff/D20165254/)

[ghstack-poisoned]

hczhu commented on torch/testing/_internal/distributed/rpc/dist_autograd_test.py:
```diff
@@ -845,6 +845,20 @@ def _verify_backwards_remote(self, tensors, context_id, local_grads, *args):
+        t2 = torch.rand((3, 3), requires_grad=True)
+        with dist_autograd.context() as context_id:
+            loss = rpc.rpc_sync(
+                "worker{}".format(self._next_rank()),
```
nit: Id' suggest defining a method in `dist_utils` like `def worker_name(rank):` to avoid using the literal string.

mrshenli reviewed (approved):
Can we also test dist_autograd does not touch `.grad`? If there is already a `.grad` tensor on the param, it should stay intact after dist backward.

pritamdamania committed:
Update on "Add test to verify dist_autograd doesn't populate .grad field."


Differential Revision: [D20165254](https://our.internmc.facebook.com/intern/diff/D20165254/)

[ghstack-poisoned]
